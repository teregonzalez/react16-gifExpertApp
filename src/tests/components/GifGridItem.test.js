import React from "react"
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('GifGridItem test', () => {

    const title = 'Un t√≠tulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(
        <GifGridItem title={ title } url={ url }/>)

    test('Debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un parrafo por el title', () => { 
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

     });

     test('Debe de tener la imagen igual al url y alt de los props', () => { 

        const img = wrapper.find('img');
        // console.log(img.props());
        expect( img.props().src ).toBe( url );
        expect( img.props().alt ).toBe( title );
         
      })

      test('Debe tener animate__fadeIn', () => { 
          
        const div = wrapper.find('div');
        const divClass = div.props().className;

        expect( divClass.includes('animate__fadeIn') ).toBe(true);
       })
})
