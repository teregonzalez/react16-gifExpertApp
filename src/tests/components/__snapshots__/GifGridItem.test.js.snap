// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem test Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un título
     
  </p>
</div>
`;
